app/assets/javascripts/backbone/views/group_modal_view.js.coffee

```diff
@@ -0,0 +1,60 @@
+class ELMO.Views.GroupModalView extends Backbone.View
+
+  el: 'body'
+
+  events:
+    'click .save': 'save'
+
+  initialize: (options) ->
+    @list_view = options.list_view
+    @mode = options.mode
+
+    # TODO: Remove once edit_group link uses url_builder
+    @edit_link = options.edit_link
+
+    if $('.group-modal').length
+      $('.group-modal').replaceWith(options.html)
+    else
+      $('body').append(options.html)
+
+    this.show()
+
+  serialize: ->
+    @form_data = $('.qing_group_form').serialize()
+
+  save: ->
+    this.serialize()
+    ELMO.app.loading(true)
+
+    if @mode == 'new'
+      this.new_group()
+    else if @mode == 'edit'
```

Comment:
If in the future there is an additional mode, then that would be the time to add in the else if? I did that in case there was one in the future.